allennlp/models/vilbert.py

```diff
@@ -714,6 +789,95 @@ def __init__(
         self.classifier = torch.nn.Linear(pooled_output_dim * 2, 2)
         self.dropout = torch.nn.Dropout(dropout)
 
+    @classmethod
+    def from_huggingface_model_name(
+        cls,
+        vocab: Vocabulary,
+        model_name: str,
+        image_feature_dim: int,
+        image_num_hidden_layers: int,
+        image_hidden_size: int,
+        combined_hidden_size: int,
+        pooled_output_dim: int,
+        image_intermediate_size: int,
+        image_attention_dropout: float,
+        image_hidden_dropout: float,
+        v_biattention_id: List[int],
+        t_biattention_id: List[int],
+        fixed_t_layer: int,
+        fixed_v_layer: int,
+        pooled_dropout: float = 0.1,
+        fusion_method: str = "sum",
+        fast_mode: bool = False,
+        with_coattention: bool = True,
+        in_batch_pairs: bool = False,
+    ):
+        transformer = AutoModel.from_pretrained(model_name)
+
+        # TODO(mattg): this is hard-coded for BERT; not sure it works for all transformers, or what
+        # to do if it fails.  We should probably pull this out into a central "transformers_util"
+        # module, or something, and just have a method that pulls out an initialized embedding layer
+        # from a huggingface model.
+        text_embeddings = deepcopy(transformer.embeddings)
+        if hasattr(transformer.config, "embedding_size"):
+            config = transformer.config
+
+            from transformers.modeling_albert import AlbertModel
+
+            if isinstance(transformer, AlbertModel):
```

Comment:
The comment above says this is just for BERT, but here you have Albert?

Reply:
Thanks, yep, I need to update that comment.